File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO2/StrategyProject/src/Sportsmen.java
public class Sportsmen
{
    private String name;
    private SportsmenStrategySwim strSwim;
    public Sportsmen(String name,SportsmenStrategySwim strSwim)
    {
        this.name = name; 
        this.strSwim = strSwim;
    }
    public String Run() { return "Running"; }
    public String Jump() { return "Jumping"; }
    public String Swim() { return strSwim.Execute(); }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2022s/ID06IO1/Strategy/Command/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    internal class Sportsmen
    {
        //private string _name;
        //public string name { get { return _name; } private set { _name = value; } }
        public string name { get; private set; }
        private SportsmenStrategyRun run;
        public Sportsmen(string name, SportsmenStrategyRun run) { this.name = name; this.run = run; }
        public string Run() { return run.Run(); }
        public string Jump() { return "Jumping"; }
        public string Swim() { return "Swiming"; }
        public string execute(SportsmenCommand command) { return name+": "+command.execute(this); }
    }
}
